t/t3431-rebase-options-compatibility.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/sh
+#
+# Copyright (c) 2019 Rohit Ashiwal
+#
+
+test_description='tests to ensure compatibility between am and interactive backends'
+
+. ./test-lib.sh
+
+# This is a special case in which both am and interactive backends
+# provide the same outputs. It was done intentionally because
+# --ignore-whitespace both the backends fall short of optimal
+# behaviour.
+test_expect_success 'setup' '
+	git checkout -b topic &&
+	q_to_tab >file <<-EOF &&
+	line 1
+	Qline 2
+	line 3
+	EOF
+	git add file &&
+	git commit -m "add file" &&
+	q_to_tab >file <<-EOF &&
+	line 1
+	new line 2
+	line 3
+	EOF
+	git commit -am "update file" &&
+	git tag side &&
+
+	git checkout --orphan master &&
+	q_to_tab >file <<-EOF &&
+	line 1
+	        line 2
+	line 3
+	EOF
+	git add file &&
+	git commit -m "add file" &&
+	git tag main &&
+	cat >output <<-EOF
+	line 1
+	new line 2
+	line 3
+	EOF
+'
+
+test_expect_success '--ignore-whitespace works with am backend' '
+	test_must_fail git rebase main side &&
+	git rebase --abort &&
+	git rebase --ignore-whitespace main side &&
+	test_cmp file output
+'
+
+test_expect_success '--ignore-whitespace works with interactive backend' '
+	test_must_fail git rebase --merge main side &&
+	git rebase --abort &&
+	git rebase --merge --ignore-whitespace main side &&
+	test_cmp file output
```

Comment:
minor nit: maybe s/output/expected, that's closer to the usual convention we're using in tests.  Also the expected bit should be the first argument in `test_cmp` to match the convention of other tests.

Reply:
Thanks for approval. I'll send it to the list in the evening.